src/sql/workbench/services/notebook/browser/notebookService.ts

```diff
@@ -129,6 +129,16 @@ export interface INotebookService {
 	 * @param isTrusted True if notebook is to be set to trusted, false otherwise.
 	 */
 	setTrusted(notebookUri: URI, isTrusted: boolean): Promise<boolean>;
+
+	/**
+	 * Event that gets fired when a new Sql query is started.
```

Comment:
It seems weird to have SQL specific events in the service API...is there any reason you couldn't just have this fire for all code cell executions and update the names to be more generic?

Reply:
@chlafreniere thoughts on this?

Reply:
@Charles-Gagnon you're right, having something more generic like onCodeCellExecutionStart event would make more sense.